src/main/io/gps.c

```diff
@@ -686,124 +1108,191 @@ void gpsInitUblox(void)
+                        ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
+                    } else {
+                        gpsData.state_position++;
+                    }
+                    break;
+                case UBLOX_INITIALIZE:
+                    ubloxSetNavRate(1, 1, 1); // throttle nav data rate to once per second, until configured
+                    break;
+                case UBLOX_ACQUIRE_MODEL:
+                    ubloxSendNAV5Message(gpsConfig()->gps_ublox_acquire_model);
+                    break;
+// *** temporarily disabled
+//                 case UBLOX_CFG_ANA:
+//                     if (gpsData.ubloxM7orAbove) { // NavX5 support existed in M5 - in theory we could remove that check
+//                         ubloxSendNavX5Message();
+//                     } else {
+//                         gpsData.state_position++;
+//                     }
+//                     break;
+                case UBLOX_MSG_DISABLE_NMEA:
+                    if (gpsData.ubloxM9orAbove) {
+                        ubloxDisableNMEAValSet();
+                        gpsData.state_position = UBLOX_MSG_RMC; // skip UBX NMEA entries - goes to RMC plus one, or SBAS
+                    } else {
+                        gpsData.state_position++;
+                    }
+                    break;
+                case UBLOX_MSG_VGS: //Disable NMEA Messages
+                    ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_VTG, 0); // VGS: Course over ground and Ground speed
+                    break;
+                case UBLOX_MSG_GSV:
+                    ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GSV, 0); // GSV: GNSS Satellites in View
+                    break;
+                case UBLOX_MSG_GLL:
+                    ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GLL, 0); // GLL: Latitude and longitude, with time of position fix and status
+                    break;
+                case UBLOX_MSG_GGA:
+                    ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GGA, 0); // GGA: Global positioning system fix data
+                    break;
+                case UBLOX_MSG_GSA:
+                    ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GSA, 0); // GSA: GNSS DOP and Active Satellites
+                    break;
+                case UBLOX_MSG_RMC:
+                    ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_RMC, 0); // RMC: Recommended Minimum data
+                    break;
+                case UBLOX_SET_SBAS:
+                    ubloxSetSbas();
+                    break;
+                case UBLOX_SET_PMS:
+                    if (gpsData.ubloxM8orAbove) {
+                        ubloxSendPowerMode();
+                    } else {
+                        gpsData.state_position++;
+                    }
+                    break;
+                case UBLOX_MSG_NAV_PVT: //Enable NAV-PVT Messages
+                    if (gpsData.ubloxM7orAbove) {
```

Comment:
(maybe flatten the if's here too)

Reply:
Thanks, I flattened similar instances, and will be grateful if you could double-check please.